extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -158,14 +211,13 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
    * @private
    */
   placeAdAfterPage_(currentPageId) {
-    // TODO(ccordry) make sure ad is loaded
-    const nextAdElement = this.adElements_[this.adElements_.length - 1];
+    const nextAdPageEl = this.adPageEls_[this.adPageEls_.length - 1];
 
-    if (!nextAdElement) {
+    if (!nextAdPageEl || !this.iscurrentAdLoaded_) {
```

Comment:
will there be race condition that a new ad is not scheduled, and an old ad gets inserted twice?

Reply:
There shouldn't be. It doesn't start loading the next ad until the previous add is placed. This question does make me realize that there maybe should be some logic in a future to move onto the next ad if we are sure that the previous ad will never load.